### DISCUSSION TITLE
cape-file completion error

### DISCUSSION THREAD

**Discussion by zijianyue** [minad/cape#113] [Q&A] [answered]
emacs-29.2 on centos7
emacs -Q, then eval below config, after that open my ~/.emacs.d/init.el and do some completion.

``` emacs-lisp
(add-to-list 'load-path "/home/gezijian/.emacs.d/elpa/corfu-20240122.2327")
(add-to-list 'load-path "/home/gezijian/.emacs.d/elpa/compat-29.1.4.4")
(add-to-list 'load-path "/home/gezijian/.emacs.d/elpa/cape-20240304.1824")

(use-package corfu
  :custom
  (corfu-auto t)
  (corfu-auto-prefix 1)
  (corfu-auto-delay 0))

(use-package cape
  :demand t
  )

(add-hook 'emacs-lisp-mode-hook
          (lambda ()
            (corfu-mode t)           
            (setq-local completion-at-point-functions
                        (list (cape-capf-super
                               #'cape-elisp-symbol
                               #'cape-dict
                               #'cape-abbrev
                               #'cape-elisp-block
                               #'cape-file
                               #'cape-dabbrev
                               )))
            
            ))
```

cape-file provided the correct candidate,

> <img width="557" alt="image" src="https://github.com/minad/cape/assets/6119928/72b566da-ad64-4ff1-8afb-19c2d30d4e84">

then I pressed enter, but it overrode the preceding dir.

> <img width="536" alt="image" src="https://github.com/minad/cape/assets/6119928/e13c57bf-e9b2-4ae1-bc42-6698e0ddea8d">

BTW， if I manually execute cape-file, the result is correct.

> <img width="555" alt="image" src="https://github.com/minad/cape/assets/6119928/f3131902-0009-4a19-8286-8c952165d1b3">

**Comment by minad** [comment]
Please try to reproduce this with cape-file directly, without cape-capf-super.

**Reply by zijianyue** [reply]
Using  cape-file directly, the result is correct, I have mentioned this in my post.

> <img width="417" alt="image" src="https://github.com/minad/cape/assets/6119928/b50a94e1-4ce7-47b5-8cac-3f2760143786">

**Reply by minad** [reply, marked as answer]
Okay, thanks. cape-capf-super doesn't support cape-file, see the documentation.

**Reply by soppelmann** [reply]
I think the confusion comes from the corfu wiki having cape-file as an example for a super-capf
https://github.com/minad/corfu/wiki#making-a-cape-super-capf-for-eglot

**Comment by zijianyue** [comment]
Dose cape-capf-super support codeium-completion-at-point?
